daemon/cmd/daemon.go

```diff
@@ -525,6 +552,7 @@ func newDaemon(ctx context.Context, cleaner *daemonCleanup,
 		apiLimiterSet:     apiLimiterSet,
 		controllers:       controller.NewManager(),
 	}
+	d.initDNSProxyContext(option.Config.DNSProxyLockSize)
```

Comment:
I realize that this is a safe place to put this initialization line, but can we instead initialize it immediately prior to handing the daemon function callback to the dnsproxy package? That way it will be easier to factor out in future if we try to further modularize the dnsproxy handling.